dapp/src/components/buySell/BuySellWidget.js

```diff
@@ -242,20 +242,30 @@ const BuySellWidget = ({
         vaultContract.autoAllocateThreshold(),
         vaultContract.rebaseThreshold(),
       ])
-
-      if (totalMintAmount.gte(allocateThreshold.mul(96).div(100))) {
-        // Define gas limit only when the amount is over threshold
-        gasLimit = gasLimits.MINT_ALLOCATE_GAS_LIMIT
-      } else if (totalMintAmount.gte(rebaseThreshold.mul(96).div(100))) {
-        gasLimit = gasLimits.MINT_REBASE_GAS_LIMIT
+      const involvingMultipleCoins = mintAddresses.length > 1
+      const aboveAllocateThreshold = totalMintAmount.gte(
+        allocateThreshold.mul(96).div(100)
+      )
+      const aboveRebaseThreshold = totalMintAmount.gte(
+        rebaseThreshold.mul(96).div(100)
+      )
+
+      if (involvingMultipleCoins) {
+        if (aboveAllocateThreshold) {
+          gasLimit = gasLimits.MINT_MULTIPLE_ALLOCATE_GAS_LIMIT
+        } else if (aboveRebaseThreshold) {
+          gasLimit = gasLimits.MINT_MULTIPLE_REBASE_GAS_LIMIT
+        } else {
+          gasLimit = gasLimits.MINT_MULTIPLE_GAS_LIMIT
+        }
       } else {
-        gasLimit = gasLimits.MINT_BASE_GAS_LIMIT
-      }
-
-      if (mintAddresses.length === 2) {
-        gasLimit += gasLimits.MINT_2_COIN_ADDITION_GAS_LIMIT
-      } else if (mintAddresses.length === 3) {
-        gasLimit += gasLimits.MINT_3_COIN_ADDITION_GAS_LIMIT
+        if (aboveAllocateThreshold) {
+          gasLimit = gasLimits.MINT_ALLOCATE_GAS_LIMIT
+        } else if (aboveRebaseThreshold) {
+          gasLimit = gasLimits.MINT_REBASE_GAS_LIMIT
+        } else {
+          gasLimit = gasLimits.MINT_GAS_LIMIT
+        }
       }
```

Comment:
☝️ I realize that nested conditionals aren't great and this is more verbose, but it's much easier to reason about than the prior base + additions thing. If I'm correct that we don't need significantly more gas for the nth coin beyond 2, then this allows us to be more precise and be more declarative with the constants.

Reply:
I think the way this is expressed is perfectly fine and readable. Sometimes (though rarely) the nested condition statements are the best approach.